.github/workflows/vendor-node-modules.yml

```diff
@@ -1,33 +1,22 @@
 name: Vendor node_modules
 
 on:
-  pull_request_target:
-    types:
-      - labeled
-  workflow_dispatch:
-    inputs:
-      pull_request:
-        description: Pull request number
-        required: true
-
-permissions:
-  contents: read
-  pull-requests: read
+  push:
+    branches:
+      - 'dependabot/**'
 
 jobs:
   vendor-node-modules:
-    if: >
-      startsWith(github.repository, 'Homebrew/') && (
-        github.event_name == 'workflow_dispatch' ||
-        contains(github.event.pull_request.labels.*.name, 'update node_modules')
-      )
     runs-on: ubuntu-latest
     steps:
       - name: Checkout
         uses: actions/checkout@v2
         with:
           persist-credentials: false
-      - uses: actions/setup-node@v2
+          fetch-depth: 0
+          token: ${{ secrets.HOMEBREW_GITHUB_PUBLIC_REPO_TOKEN }}
```

Comment:
Is there a reason we need this?

Reply:
For pushing back the changes.

Reply:
`GITHUB_TOKEN` can push to the PR brach, no?

Reply:
Not if your CI runs long enough

Reply:
> For pushing back the changes.

Even though we have `persist-credentials: false`?

Ideally this would only be set after `npm install`, but I just realised that I'm not sure if Dependabot actually has access to secrets. Setting `permissions` does work but having global write privileges isn't ideal.

This probably needs to be `pull_request_target` with down-scoped `permissions` and then use the public repo token secret on push.